### PR TITLE
chore(project): switch to Java 11 source code level

### PR DESCRIPTION
**Pull request by yanavasileva**
and build release parent with JDK 11

https://github.com/camunda/camunda-bpm-platform/issues/3690

### CODE REVIEW

**Comment by yanavasileva** [comment]
Decisions (I will move them to the ticket later):
* use maven 3.8 for building the project
* use `release` configuration property to define the java source code level. Therefore bump the `maven-compiler-plugin` to latest 
* bump maven bundle plugin and javadoc plugin to next highest stable version working with JDK 11

File: pom.xml [hunk 1]
```diff
@@ -24,11 +24,14 @@
...
 
-    <!-- we use java 1.8 for our projects -->
-    <version.java>1.8</version.java>
+    <!-- we use Java 11 for our projects -->
+    <version.java>11</version.java>
 
     <plugin.version.license>3.0</plugin.version.license>
     <plugin.version.nexus-staging>1.6.13</plugin.version.nexus-staging>
+    <plugin.version.compiler>3.6.2</plugin.version.compiler>
+    <plugin.version.javadoc>3.0.1</plugin.version.javadoc>
```
Comment (mboskamp) [inline pom.xml:33]: I found that we define the same version in the [platform pom.xml](https://github.com/camunda/camunda-bpm-platform/blob/2204b2830bef534796c42fb8df4f28673c5e406f/pom.xml#L36). We can remove it there. Maybe there are other projects that we can simplify by removing the version there?

**Review by mboskamp — APPROVED** [review]
👍 Looks good! Nice catch with the Maven version in the CI file. I approved the PR already since it looks good in general. For the platform and possibly other projects that specify a javadoc version, we need another PR anyway. Feel free to go ahead with this one :)

File: pom.xml (hunk 1 above)
Comment (yanavasileva) [reply pom.xml:33]: 👍

**Pull request merged**